qcengine/programs/entos.py

```diff
@@ -0,0 +1,238 @@
+"""
+Calls the entos executable.
+"""
+
+from typing import Any, Dict, Optional, List
+
+from qcelemental.models import Result, FailedOperation
+from ..util import execute, popen
+from qcelemental.util import which, safe_version, parse_version, which_import
+
+from .model import ProgramHarness
+
+
+class EntosHarness(ProgramHarness):
+    _defaults = {
+        "name": "entos",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    def __init__(self, **kwargs):
+        super().__init__(**{**self._defaults, **kwargs})
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which('entos', return_bool=True, raise_error=raise_error, raise_msg='Please install via XXX')
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        which_prog = which('entos')
+        if which_prog not in self.version_cache:
+            with popen([which_prog, '--version']) as exc:
+                exc["proc"].wait(timeout=5)
```

Comment:
We should bump timeout to ~15 seconds. There are some very slow supercomputers out there.